app/views/school_groups/recent_usage.html.erb

```diff
@@ -27,40 +28,75 @@
         <th><%= t('common.labels.last_week') %></th>
         <th><%= t('common.labels.last_year') %></th>
       <% end %>
-      <th></th>
     </tr>
   </thead>
   <tbody>
     <% @school_group.schools.visible.order(:name).each do |school| %>
-      <% recent_usage = school.recent_usage %>
       <tr>
-        <% if @fuel_types.include?(:electricity) %>
-          <td class="<%= recent_usage&.electricity&.week&.message_class %>" data-order="<%= recent_usage&.electricity&.week&.change&.gsub('%','')&.to_f || '0' %>">
-            <%= up_downify(recent_usage&.electricity&.week&.change || 'n/a') %>
-          </td>
-          <td class="<%= recent_usage&.electricity&.year&.message_class %>" data-order="<%= recent_usage&.electricity&.year&.change&.gsub('%','')&.to_f || '0'  %>">
-            <%= up_downify(recent_usage&.electricity&.year&.change || 'n/a') %>
-          </td>
-        <% end %>
-        <% if @fuel_types.include?(:gas) %>
-          <td class="<%= recent_usage&.gas&.week&.message_class %>" data-order="<%= recent_usage&.gas&.week&.change&.gsub('%','')&.to_f || '0' %>">
-            <%= up_downify(recent_usage&.gas&.week&.change || 'n/a') %>
-          </td>
-          <td class="<%= recent_usage&.gas&.year&.message_class %>" data-order="<%= recent_usage&.gas&.year&.change&.gsub('%','')&.to_f || '0' %>">
-            <%= up_downify(recent_usage&.gas&.year&.change || 'n/a') %>
-          </td>
-        <% end %>
-        <% if @fuel_types.include?(:storage_heaters) %>
-          <td class="<%= recent_usage&.storage_heaters&.week&.message_class %>" data-order="<%= recent_usage&.storage_heaters&.week&.change&.gsub('%','')&.to_f || '0' %>">
-            <%= up_downify(recent_usage&.storage_heaters&.week&.change || 'n/a') %>
-          </td>
-          <td class="<%= recent_usage&.storage_heaters&.year&.message_class %>" data-order="<%= recent_usage&.storage_heaters&.year&.change&.gsub('%','')&.to_f || '0' %>">
-            <%= up_downify(recent_usage&.storage_heaters&.year&.change || 'n/a') %>
-          </td>
-        <% end %>
         <td>
           <%= link_to school.name, school_path(school) %>
         </td>
+        <% if school.data_enabled %>
```

Comment:
Perhaps some of these nested ifs could be extracted to a single partial?  e.g. replace lines 34-101 with
```
    <% @school_group.schools.visible.order(:name).each do |school| %>
      <tr>
        <td>
          <%= link_to school.name, school_path(school) %>
        </td>
        <% if school.data_enabled %>
          <% recent_usage = school.recent_usage %>
          <%= render 'recent_usage_table_row', fuel_type: :electricity, recent_usage: recent_usage.electricity if  @fuel_types.include?(:electricity) %>
          <%= render 'recent_usage_table_row', fuel_type: :gas, recent_usage: recent_usage.gas if @fuel_types.include?(:gas) %>
          <%= render 'recent_usage_table_row', fuel_type: :storage_heaters, recent_usage: recent_usage.storage_heaters if @fuel_types.include?(:storage_heaters) %>
        <% else %>
          <% (@fuel_types.reject{|f| f == :solar_pv}.length * 2).times do %>
            <td class="text-right" data-order="'0'">-</td>
          <% end %>
        <% end %>
      </tr>
    <% end %>
```
and then a partial `_recent_usage_table_row.html.erb` that looks like
```
<% if recent_usage&.week&.has_data %>
  <td class="text-right <%= recent_usage&.week&.message_class %>"
      data-order="<%= recent_usage&.week&.change&.gsub('%','')&.to_f || '0' %>">
    <%= up_downify(recent_usage&.week&.change) %>
  </td>
<% else %>
  <td class="text-right" data-order="0">-</td>
<% end %>
<% if recent_usage&.year&.has_data %>
  <td class="text-right <%= recent_usage&.year&.message_class %>"
      data-order="<%= recent_usage&.year&.change&.gsub('%','')&.to_f || '0'  %>">
    <%= up_downify(recent_usage&.year&.change) %>
  </td>
<% else %>
  <td class="text-right" data-order="0">-</td>
<% end %>
```

Reply:
I may do that in a later PR, but I want to get some feedback on the current table structure first.